test/controllers/comment_controller_test.exs

```diff
@@ -34,149 +32,70 @@ defmodule CodeCorps.CommentControllerTest do
   describe "show" do
     test "shows chosen resource", %{conn: conn} do
       comment = insert(:comment)
-
-      path = conn |> comment_path(:show, comment)
-      conn = conn |> get(path)
-
-      data = json_response(conn, 200)["data"]
-
-      assert data["id"] == "#{comment.id}"
-      assert data["type"] == "comment"
-      assert data["attributes"]["body"] == comment.body
-      assert data["attributes"]["markdown"] == comment.markdown
-      assert data["relationships"]["user"]["data"]["id"] == "#{comment.user_id}"
-      assert data["relationships"]["task"]["data"]["id"] == "#{comment.task_id}"
+      conn
+      |> request_show(comment)
+      |> json_response(200)
+      |> Map.get("data")
+      |> assert_result_id(comment.id)
     end
 
     test "renders 404 when id is nonexistent", %{conn: conn} do
-      path = conn |> comment_path(:show, -1)
-      assert conn |> get(path) |> json_response(404)
+      assert conn |> request_show(:not_found) |> json_response(404)
     end
   end
 
   describe "create" do
     @tag :authenticated
     test "creates and renders resource when data is valid", %{conn: conn, current_user: current_user} do
       task = insert(:task)
-
-      payload =
-        build_payload
-        |> put_attributes(@valid_attrs)
-        |> put_relationships(current_user, task)
-
-      path = conn |> comment_path(:create)
-      conn = conn |> post(path, payload)
-
-      assert json_response(conn, 201)["data"]["id"]
+      attrs = @valid_attrs |> Map.merge(%{task: task, user: current_user})
```

Comment:
We use `Map.merge` instead of `put_relationships`, but I would prefer to refactor `put_relationships` instead.